libtransmission/announcer.cc

```diff
@@ -1741,22 +1738,25 @@ static void scrapeAndAnnounceMore(tr_announcer* announcer)
      * we can work through that queue much faster than announces
      * (thanks to multiscrape) _and_ the scrape responses will tell
      * us which swarms are interesting and should be announced next. */
-    multiscrape(announcer, &scrapeMe);
+    multiscrape(announcer, scrape_me);
 
     /* Second, announce what we can. If there aren't enough slots
      * available, use compareAnnounceTiers to prioritize. */
-    int n = MIN(tr_ptrArraySize(&announceMe), MAX_ANNOUNCES_PER_UPKEEP);
-    for (int i = 0; i < n; ++i)
+    if (announce_me.size() > MAX_ANNOUNCES_PER_UPKEEP)
+    {
+        std::partial_sort(
+            std::begin(announce_me),
+            std::begin(announce_me) + MAX_ANNOUNCES_PER_UPKEEP,
+            std::end(announce_me),
+            [](tr_tier const* a, tr_tier const* b) { return compareAnnounceTiers(a, b) < 0; });
```

Comment:
That's the only place where `compareAnnounceTiers()` is used, maybe return `bool` (if less) from it right away; then lambda won't even be needed.

Reply:
Here and elsewhere: spelling out parameter type isn't necessary (unless you want to).
```suggestion
            [](auto const* a, auto const* b) { return compareAnnounceTiers(a, b) < 0; });
```

Reply:
I had made it explicit out of fear that the older compilers in CI wouldn't handle it (e.g. gcc 5.5.0), but you're right. Let's give it a try.

Fixed ecd14048b7f4a2741646eb10d54d8064a902585a